Add debian creation functionality

padmashree9110 opened the pull request:
Adding debian creation functionality and udpated folder structure to ease the same

padmashree9110 committed:
Add debian creation functionality

Adding debian creation functionality and udpated folder structure to ease the same

padmashree9110 committed:
Code restructuring to follow ROS standards

Commit:
Update instructions to create debian package

Update instructions to create debian packages and improved docs.

anitha-suresh-zz reviewed (approved):
Tested. All examples are working.

Commit:
Improve Makefile that creates debian packages

We were hard-coding object directories created by debian build.
It is not right way to keep hard-coded directory name in `.install` file.
Now, Makefile composes debian `.install` for the package whom make is issued for.

shakthi-prashanth-m commented:
@anitha-suresh Kindly review the latest [commit](https://github.com/intel-aero/sample-apps/pull/8/commits/edfed281f6e469c910d0632fbfea41c4b4933d33). We can merge once it is reviewed.
